plugin/trino-hive/src/main/java/io/trino/plugin/hive/CachingDirectoryLister.java

```diff
@@ -87,20 +95,69 @@ private static SchemaTablePrefix parseTableName(String tableName)
         }
         return new SchemaTablePrefix(schema, table);
     }
-
+    
+    /**
+     * Execute and return the result from {@link #listExecuter()} incorporated with a timer. 
+     * If listExecuter does not end within a certain amount of time (defined by 
+     * the field variable fileListingTimeout) the Thread running listExecuter will 
+     * be interrupted and a RuntimeException will be thrown. 
+     * 
+     * @param fs FileSystem, see: {@link org.apache.hadoop.fs.FileSystem}
+     * @param table Table, see: {@link io.trino.plugin.hive.metastore.Table}
+     * @param path  Path, see: {@link org.apache.hadoop.fs.Path}
+     * @return an iterator that runs over FileStatus-Object that inclues a file's block 
+     * locations
+     * @throws IOException
+     * @throws RuntimeException
```

Comment:
I am a bit undecided about javadoc for this project. It is good but I did not find the rest of the code to be using it. But do you think it is beneficial enough to keep it?

Reply:
I personally think it's better to have than not to have. At least I would be happy if the function have descriptions =).

Reply:
I also think that comments in general are better, but the fact is that the project does not have comments in the code, so I think it would be better to follow that style, even if you disagree with it.

Reply:
I agree with the others about the comments, very good to have but the project does not have in its style.